### PR TITLE
Consider disk space used by fusion indexes beyond current fusion index or

### PR DESCRIPTION
**Pull request by toregge**
indexes before current fusion index as transient disk space.

@geirst : please review

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: searchcorespi/src/vespa/searchcorespi/index/index_disk_dir.h [hunk 1]
```diff
@@ -29,6 +29,8 @@ class IndexDiskDir {
         return (_id == rhs._id) && (_fusion == rhs._fusion);
     }
     bool valid() const noexcept { return _id != 0u; }
+    bool get_fusion() const noexcept { return _fusion; }
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchcorespi/src/vespa/searchcorespi/index/index_disk_dir.h]: Done

File: searchcorespi/src/tests/index/active_disk_indexes/active_disk_indexes_test.cpp [hunk 2]
```diff
@@ -48,6 +70,90 @@ TEST_F(ActiveDiskIndexesTest, is_active_returns_false_for_bad_name)
...
+    EXPECT_TRUE(remove(fusion1));
+    setActive("index.fusion.1", 0);
+    EXPECT_FALSE(remove(fusion1));
+    notActive("index.fusion.1");
+    EXPECT_TRUE(remove(fusion1));
+}
+
+TEST_F(ActiveDiskIndexesTest, basic_get_transient_size_works)
+{
+    setActive("index.fusion.1", 1000000);
```
Comment (geirst) [inline searchcorespi/src/tests/index/active_disk_indexes/active_disk_indexes_test.cpp:88]: Please add comment describing which index collection setup this represents.

Comment (toregge) [reply searchcorespi/src/tests/index/active_disk_indexes/active_disk_indexes_test.cpp]: Done.

File: searchcorespi/src/tests/index/active_disk_indexes/active_disk_indexes_test.cpp [hunk 3]
```diff
@@ -48,6 +70,90 @@ TEST_F(ActiveDiskIndexesTest, is_active_returns_false_for_bad_name)
...
+    notActive("index.fusion.2");
+    {
+        SCOPED_TRACE("index.fusion.1 after remove of index.fusion.2");
+        assert_transient_size(0, fusion1);
+    }
+}
+
+TEST_F(ActiveDiskIndexesTest, dynamic_get_transient_size_works)
+{
+    setActive("index.fusion.1", 1000000);
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchcorespi/src/tests/index/active_disk_indexes/active_disk_indexes_test.cpp]: Done.

File: searchcorespi/src/vespa/searchcorespi/index/activediskindexes.cpp [hunk 4]
```diff
@@ -44,4 +51,66 @@ bool ActiveDiskIndexes::isActive(const string &index) const {
...
+{
+    if (!index_disk_dir.valid() || !index_disk_dir.is_fusion_index()) {
+        return 0u;
+    }
+    uint64_t transient_size = 0u;
+    std::vector<IndexDiskDir> deferred;
+    {
+        std::lock_guard lock(_lock);
+        for (auto &entry : _active) {
+            if (entry.first < index_disk_dir) {
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchcorespi/src/vespa/searchcorespi/index/activediskindexes.cpp]: Done.

File: searchcorespi/src/vespa/searchcorespi/index/activediskindexes.h [hunk 5]
```diff
@@ -11,6 +11,7 @@ namespace searchcorespi::index {
 
 class IndexDiskDir;
 class IndexDiskDirActiveState;
+class IndexDiskLayout;
 
 /**
  * Class used to keep track of the set of active disk indexes in an index maintainer.
```
[the remark this answers is not in the document]
Comment (toregge) [reply searchcorespi/src/vespa/searchcorespi/index/activediskindexes.h]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**